src/core/backbone_router/local.cpp

```diff
@@ -242,6 +245,134 @@ void Local::UpdateBackboneRouterPrimary(Leader::State aState, const BackboneRout
     return;
 }
 
+otError Local::GetDomainPrefix(NetworkData::OnMeshPrefixConfig &aDomainPrefixConfig)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength > 0, error = OT_ERROR_NOT_FOUND);
+
+    aDomainPrefixConfig = mDomainPrefixConfig;
+
+exit:
+    return error;
+}
+
+otError Local::RemoveDomainPrefix(const otIp6Prefix &aPrefix)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength > 0, error = OT_ERROR_NOT_FOUND);
+    VerifyOrExit(otIp6PrefixMatch(&mDomainPrefixConfig.mPrefix.mPrefix, &aPrefix.mPrefix) >=
+                     mDomainPrefixConfig.mPrefix.mLength,
+                 error = OT_ERROR_NOT_FOUND);
+
+    memset(&mDomainPrefixConfig, 0, sizeof(mDomainPrefixConfig));
+
+    if (IsEnabled())
+    {
+        SuccessOrExit(error =
+                          Get<NetworkData::Local>().RemoveOnMeshPrefix(aPrefix.mPrefix.mFields.m8, aPrefix.mLength));
+        IgnoreReturnValue(Get<NetworkData::Local>().SendServerDataNotification());
+    }
+
+exit:
+    return error;
+}
+
+otError Local::SetDomainPrefix(const NetworkData::OnMeshPrefixConfig &aDomainPrefixConfig)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aDomainPrefixConfig.mPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+
+    if (IsEnabled())
+    {
+        Get<NetworkData::Local>().RemoveOnMeshPrefix(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8,
+                                                     mDomainPrefixConfig.mPrefix.mLength);
+    }
+
+    mDomainPrefixConfig = aDomainPrefixConfig;
+
+    if (IsEnabled())
+    {
+        uint8_t flags = 0;
+
+        if (mDomainPrefixConfig.mPreferred)
```

Comment:
This seems to share a lot of code with `otBorderRouterAddOnMeshPrefix()`. Maybe `NetworkData::Local::AddOnMeshPrefix()` should take `OnMeshPrefixConfig` and include this code to avoid duplication?

Reply:
updated as your suggestion.